### PR TITLE
DVDVideoCodecAmlogic: Add VP9 HW decoding support

### PR DESCRIPTION
**Pull request by kszaq**
Adds VP9 hardware decoding for AMLCodec on supported devices.
`vp9_update_frame_header()` is adapted from Amlogic code, like every other in AMLCodec.cpp

## How Has This Been Tested?
LibreELEC community builds for S905X and S912 devices with Kodi v16, v17 and master.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by peak3d** [comment]
@kszaq where is it defined?

/dev/xbmc/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp: In function ‘vformat_t codecid_to_vformat(AVCodecID)’:
/dev/xbmc/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp:420:16: error: ‘VFORMAT_VP9’ was not declared in this scope
       format = VFORMAT_VP9;

**Comment by kszaq** [comment]
@peak3d You need to use the latest libamcodec. It available in Amlogic buildroot or if you want to build LE you can use this patch: https://github.com/kszaq/LibreELEC.tv/commit/0eee158da611cf9314dffb0c299df96a72699b9c#diff-5e054f75cd5227ef84ceca9d7bdead69

**Comment by kszaq** [comment]
@peak3d I added `#define VFORMAT_VP9` in case it is undefined - I noticed that this is needed if you're using an older version of amcodec, e.g. for S8xx devices.

**Comment by lrusak** [comment]
Please follow the code guidlines 🐱 https://github.com/xbmc/xbmc/blob/master/CODING_GUIDELINES.dox#L94

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp [hunk 1]
```diff
@@ -902,6 +908,143 @@ static int hevc_write_header(am_private_t *para, am_packet_t *pkt)
...
+    fdata[9] = 0;
+    fdata[10] = 0;
+    fdata[11] = 1;
+    fdata[12] = 'A';
+    fdata[13] = 'M';
+    fdata[14] = 'L';
+    fdata[15] = 'V';
+    framesize -= 4;/*del 4 to real framesize for check.....*/
+
+    if (!old_header) {
```
Comment (peak3d) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp]: @kszaq only this one (new line) and we're ready to go....

**Comment by peak3d** [comment]
@kszaq can you pls. check for this existance as well?

dec_type = VIDEO_DEC_FORMAT_VP9;

**Comment by kszaq** [comment]
@peak3d Done.

**Review by peak3d — APPROVED** [review]

**Comment by peak3d** [comment]
thx, @kszaq !

**Pull request merged**